### PR TITLE
PRDO-619: Add Deck Credit Info Drawer

### PR DESCRIPTION
**Pull request by neel-ds**
- Description
Add info drawer and entry cost badge on deck preview card

- What are the steps to test that this code is working?
open a deck

- Screen shots or recordings for UI changes
Screenshot 2024-12-27 at 8 02 09 PM
Screenshot 2024-12-27 at 8 02 20 PM

### CODE REVIEW

**Comment by yash431garg** [comment]
Can you check the spacing in pop up? 
Screenshot 2024-12-30 at 8 37 55 AM
tps://github.com/user-attachments/assets/de7cd8ce-ced8-4a17-81db-9a1aeb387150" />

**Review by yash431garg — CHANGES_REQUESTED** [review]
Can you check the spacing in the pop up? 
Screenshot 2024-12-30 at 8 37 55 AM
Screenshot 2024-12-30 at 8 40 21 AM

File: app/screens/DeckScreens/DeckScreen.tsx [hunk 1]
```diff
@@ -56,11 +56,11 @@ const DeckScreen = ({
       {!isDeckStarted ? (
         <div className="flex flex-col gap-4 h-full w-full">
           {CREDIT_COST_FEATURE_FLAG && (
-            <div className="rounded-[56px] bg-chomp-blue-light text-xs text-gray-900 p-2 w-fit">
+            <div className="rounded-[56px] bg-chomp-blue-light text-xs text-gray-900 font-medium px-2 py-1 w-fit">
               {totalCredits > (deckCost ?? 0) ? (
-                <span className="opacity-50">Balance </span>
+                <span className="opacity-60">Balance </span>
```
Comment (yash431garg) [inline app/screens/DeckScreens/DeckScreen.tsx:61]: In figma opacity is 50

Comment (yash431garg) [inline app/screens/DeckScreens/DeckScreen.tsx:61]: In figma design the opacity is 50.

**Comment by neel-ds** [comment]
> Can you check the spacing in the pop up? Screenshot 2024-12-30 at 8 37 55 AM Screenshot 2024-12-30 at 8 40 21 AM

Adjusted the items gap

**Pull request merged**